### PR TITLE
Turned exercise 022 into practice

### PR DESCRIPTION
**Pull request by divyankachaudhari**
Addresses #2198. Turns [Exercise 022](https://github.com/ocaml/ocaml.org/blob/main/data/exercises/022_create_list.md) into practice.

Added two extra test cases:
- Descending order
- Same number

The solution in the markdown has two solutions: a non-tail recursive and a tail recursive. So, with #2210 as precedent, I added one solution normally and I added the other one in comments.

### CODE REVIEW

File: practice/022/answer/impl.ml
Comment (cuihtlauac) [inline practice/022/answer/impl.ml:1]: Turn the tail recursive implementation into an `answer2` folder.

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

File: practice/022/answer/impl.ml
Comment (divyankachaudhari) [reply practice/022/answer/impl.ml:1]: Done.

**Comment by divyankachaudhari** [comment]
Added two solutions, one in `answer` and tail-recursive one in `answer2`.

Both `dune test --root . answer` and `dune test --root . answer2` work.

**Comment by divyankachaudhari** [comment]
@cuihtlauac Could you please review this again? Thank you.

**Review by cuihtlauac — APPROVED** [review]
Thanks @divyankachaudhari, sorry for the delay in reviewing

**Review by cuihtlauac — APPROVED** [review]

**Pull request merged**